### PR TITLE
Add an OnSend handler.

### PR DESCRIPTION
**Pull request by rrcollier**
Similar to the OnReceive handler but called when a packet is sent. Reuses the Packet type, but does not set Ttl or Rtt.

This was originally accidentally included in PR 104. Now seperated out.

### CODE REVIEW

File: ping.go [hunk 1]
```diff
@@ -587,6 +590,17 @@ func (p *Pinger) sendICMP(conn *icmp.PacketConn) error {
...
+
+		handler := p.OnSend
+		if handler != nil {
+			outPkt := &Packet{
+				Nbytes: len(msgBytes),
+				IPAddr: p.ipaddr,
+				Addr:   p.addr,
+				Seq:    p.sequence - 1,
+			}
+			handler(outPkt)
```
Comment (CHTJonas) [inline ping.go:602]: I did wonder whether this should be asynchronous `go handler(outPkt)` but it matches the other `OnRecv` event handlers we have in the code. We can look at this in a future PR perhaps.

**Review by CHTJonas — APPROVED** [review]
~LGTM 👍~ Edit: see review below.

File: ping.go [hunk 2]
```diff
@@ -587,6 +590,17 @@ func (p *Pinger) sendICMP(conn *icmp.PacketConn) error {
...
+		}
```
Comment (CHTJonas) [inline ping.go:603]: If you swap around the order of execution it avoid doing an unnecessary `p.sequence - 1`:

```suggestion
		handler := p.OnSend
		if handler != nil {
			outPkt := &Packet{
				Nbytes: len(msgBytes),
				IPAddr: p.ipaddr,
				Addr:   p.addr,
				Seq:    p.sequence,
			}
			handler(outPkt)
		}

		p.PacketsSent++
		p.sequence++
```

**Review by CHTJonas — CHANGES_REQUESTED** [review]

File: ping.go (hunk 1 above)
Comment (rrcollier) [reply ping.go:602]: Yes, I modeled the behaviour after the OnRecv handler.

**Review by SuperQ — APPROVED** [review]
LGTM

**Review by CHTJonas — APPROVED** [review]

**Pull request merged**